modules/repo_sync/src/Service/Client.php

```diff
@@ -189,11 +194,14 @@ protected function withCurl(string $method, string $url, array $headers, ?callab
       $callback($ch);
     }
 
-    $rawresult = curl_exec($ch);
-    $header_size = curl_getinfo($ch, CURLINFO_HEADER_SIZE);
-    $result = substr($rawresult, $header_size);
-    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-
+    if ($rawresult = curl_exec($ch)) {
```

Comment:
This can be rewritten as:

    if (!($rawresult = curl_exec($ch))) {
      throw new Exception(...);
    }

I think it would also make sense to create an exception type for this, so it would be possible to store related metadata.